src/connectors/onchain_events/mod.rs

```diff
@@ -581,46 +618,28 @@ impl Subscriber {
             latest_block_in_db,
             "Starting l2 events subscription"
         );
-        let historical_sync_start_block = latest_block_in_db.max(self.start_block_number);
-        let historical_sync_stop_block =
-            latest_block_on_chain.min(self.stop_block_number.unwrap_or(latest_block_on_chain));
-
-        self.sync_historical_events(historical_sync_start_block, historical_sync_stop_block)
-            .await?;
+        match self.start_block_number {
+            None => {
+                self.sync_live_events(latest_block_on_chain).await?;
+            }
+            Some(start_block_number) => {
+                let historical_sync_start_block = latest_block_in_db.max(start_block_number);
```

Comment:
We should also do a `max(start_block_number, FIRST_BLOCK)` just to make sure we don't accidentally scan from 0 for blocks. We should never have to look at bloks older than FIRST_BLOCK